#==================================================================================================
# Patch to fix ARM detection, for CMake. Upstream already has these for autotools-based builds,
# but hasn't added these for CMake yet.
#
# See: https://trac.macports.org/ticket/67572
#==================================================================================================
--- cmake/DefineTargetVariables.cmake.orig	2023-06-06 11:49:00.000000000 -0400
+++ cmake/DefineTargetVariables.cmake	2023-06-06 11:49:37.000000000 -0400
@@ -7,7 +7,7 @@
   check_cxx_source_compiles("int main() { return __s390__; }" s390)
   check_cxx_source_compiles("int main() { return __PPC64__; }" PPC64)
   check_cxx_source_compiles("int main() { return __x86_64__; }" x86_64)
-  check_cxx_source_compiles("int main() { return __arm__; }" ARM)
+  check_cxx_source_compiles("int main() { return __arm64__; }" ARM)
   check_cxx_source_compiles("int main() { return __FreeBSD__; }" FreeBSD)
   check_cxx_source_compiles("int main() { return __MINGW__; }" MINGW)
   check_cxx_source_compiles("int main() { return __linux; }" LINUX)
--- cmake/PCFromUContext.cmake.orig	2023-06-06 11:40:58.000000000 -0400
+++ cmake/PCFromUContext.cmake	2023-06-06 11:45:52.000000000 -0400
@@ -39,7 +39,9 @@
         "uc_mcontext->ss.rip"  # OS X (x86_64)
         "uc_mcontext->__ss.__rip"  # OS X (>=10.5 [untested])
         "uc_mcontext->ss.srr0"  # OS X (ppc, ppc64 [untested])
-        "uc_mcontext->__ss.__srr0")  # OS X (>=10.5 [untested])
+        "uc_mcontext->__ss.__srr0"  # OS X (>=10.5 [untested])
+        "uc_mcontext->__ss.__pc"  # OS X (arm64)
+    )
 
     set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE=1)
     if(HAVE_CYGWIN_SIGNAL_H)
